mypy.ini

```diff
@@ -0,0 +1,4 @@
+[mypy]
+
+[mypy-allennlp.tests*]
+ignore_errors = True
```

Comment:
Why is this necessary? Did `mypy` previously not run on our tests?

Reply:
Yes, mypy was previously not being run on the testing code --- I thought this was intentional?

Reply:
Seems like you made the right change in this PR then.  I'm not sure what we want long term.

Reply:
@nelson-liu, do the mypy errors on the tests look helpful?  I'd guess that we don't care about them, because it's not as important to have clear type annotations in the tests, but if they look useful we could reconsider.

Reply:
@matt-gardner here are the mypy errors: https://gist.github.com/nelson-liu/f94a96265973a390066f45399718fe5e . I guess they're not entirely useless? Perhaps we should address in another PR though, this one is already huge.

Reply:
Hmm, yeah, ok, I'll open that as an issue.  Thanks for doing all of this!